FirebaseFunctions/Sources/HTTPSCallable.swift

```diff
@@ -79,7 +79,7 @@ open class HTTPSCallable: NSObject {
                                                    completion: @escaping (HTTPSCallableResult?,
                                                                           Error?) -> Void) {
     if #available(iOS 13, macCatalyst 13, macOS 10.15, tvOS 13, watchOS 7, *) {
-      Task {
+      Task { @MainActor in
         do {
           let result = try await call(data)
```

Comment:
Should only the completion be on the MainActor? Could the `call` lock it up?

Reply:
I thought the same thing. I dropped a breakpoint in `call` and it was _off_ the main thread.

Reply:
So the `call` will execute off of the main thread and won't block. We could be more precise and wrap the completion specifically which is arguably more defensive in case the logic in this scope grows and we want to minimize the work that occurs on the main thread.

Reply:
Why not be more precise? That would also be consistent with how Auth typically does completions.

Reply:
I did a more precise approach in 9689fd8. But then decided to try the approach where we define the completion param as synchronized to the main actor (bdf79b3). This latter commit is better since the completion will be called on the main actor without needing to dispatch to it with `DispatchQueue.main.async { ... }` or `await MainActor.run { ... }`.

Reply:
Previously marking the completion would lead to warnings when using Swift 5 x Xcode 15. In Xcode 16 though, I'm not seeing them issue warnings with Swift 5 projects. for Swift 6 mode, it makes it more restrictive as the closure is typed checked to the main actor.